keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/EnvironmentArtifactsSummary.kt

```diff
@@ -0,0 +1,16 @@
+package com.netflix.spinnaker.keel.api
+
+data class EnvironmentArtifactsSummary(
+  val name: String,
+  val artifacts: Collection<ArtifactVersions>
+)
+
+data class ArtifactVersions(
+  val name: String,
+  val type: ArtifactType,
+  val versions: Map<PromotionStatus, List<String>>
+)
+
+enum class PromotionStatus {
+  PREVIOUS, CURRENT, DEPLOYING, PENDING
```

Comment:
I think you might need another state here. 

There are potentially many artifacts that were approved for an environment but not deployed to it. How are you representing that state?

Reply:
Yeah, right now we're not and they show up as `DEPLOYING` which is a problem I only noticed when I pointed it at real data.